packages/server/src/asyncQuestion/asyncQuestion.controller.ts

```diff
@@ -44,6 +45,62 @@ export class asyncQuestionController {
     private mailService: MailService,
   ) {}
 
+  @Post('comment')
+  @Roles(Role.STUDENT, Role.TA, Role.PROFESSOR)
+  async replyToQuestion(
+    @Body() body: any,
+    @User() user: UserModel,
+    @Res() res: Response,
+  ): Promise<Response> {
+    const { questionId, commentText } = body;
+    const question = await AsyncQuestionModel.findOne({
+      where: { id: questionId },
+      relations: ['course', 'creator'],
+    });
+
+    if (!question) {
+      res
+        .status(HttpStatus.NOT_FOUND)
+        .send({ message: ERROR_MESSAGES.questionController.notFound });
+      return;
+    }
+
+    // check if user has commented within 2 minutes
+    const lastComment = await AsyncQuestionCommentsModel.findOne({
+      where: { creator: user, question },
+      order: { createdAt: 'DESC' },
+    });
+
+    if (lastComment) {
+      const timeDiff = new Date().getTime() - lastComment.createdAt.getTime();
+      if (timeDiff < 30000) {
+        res.status(HttpStatus.BAD_REQUEST).send({
+          message: ERROR_MESSAGES.questionController.createQuestion.tooFast,
+        });
+        return;
+      }
+    }
+
+    const comment = await AsyncQuestionCommentsModel.create({
+      commentText,
+      creator: user,
+      question,
+      createdAt: new Date(),
+    }).save();
+
+    const updatedQuestion = await AsyncQuestionModel.findOne({
+      where: { id: questionId },
+      relations: ['creator', 'taHelped', 'votes', 'comments'],
```

Comment:
I can't leave a comment on the right lines but there's some updatedQuestion queries in the functions below that you will want to update to include 'comments'
relations: ['creator', 'taHelped', 'votes'], -> relations: ['creator', 'taHelped', 'votes', 'comments'],